tests/test_decorators.py

```diff
@@ -378,22 +387,192 @@ def function_b():
+        if "duration" in message_dict:
+            durations.append(message_dict["duration"])
+    assert all(0.011 == pytest.approx(duration, abs=0.005) for duration in durations)
+
+
+def test_timer_as_context_manager(log_store, logger) -> None:
     logger.addHandler(log_store)
-    with Timer(logger=logger_console.warning) as time:
-        concat_two_strings_warning('a', 'b')
+    with Timer(logger=logger_console.warning):
+        concat_two_strings_warning("a", "b")
     all_messages = " ".join(log_store.messages["warning"])
-    assert 'exception' not in all_messages
+    assert "exception" not in all_messages
     log_store.reset()
 
-    with Timer(logger=logger_console.warning, suppress_exceptions=True) as time:
+    with Timer(logger=logger_console.warning, suppress_exceptions=True):
         raise Exception
     all_messages = " ".join(log_store.messages["warning"])
-    assert 'exception' in all_messages
+    assert "exception" in all_messages
     log_store.reset()
+
+
+class TestThreadContextLogger:
+    @staticmethod
+    @pytest.mark.parametrize(
+        "message, expected_message",
+        [
+            # nothing happens when the message is a plain string
+            (
+                "hello",
+                "hello",
+            ),
+            # add a context if the message is a dict
+            (
+                {"message": "hello"},
+                {"message": "hello", "ctx": 123},
+            ),
+        ],
+    )
+    def test_thread_context_logger_as_context_manager(
+        log_store: MockLoggingHandler,
+        logger: Logger,
+        message: Any,
+        expected_message: Any,
+    ) -> None:
+        logger.addHandler(log_store)
+
+        logger.info(message)
+
+        with ThreadContextLogger(logger=logger, context_dict={"ctx": 123}):
+            logger.info(message)
+
+        logger.info(message)
+
+        logged_messages: List[str] = log_store.messages["info"]
+        assert len(logged_messages) == 3
+
+        for i, logged_message in enumerate(logged_messages):
+            try:
+                logged_message = eval(logged_message)
+            except NameError:
+                pass
+            if i == 1:
+                # message affected with the thread context logger
+                assert logged_message == expected_message
+            else:
+                # message before/after the thread context logger
+                assert logged_message == message
+
+        log_store.reset()
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        "message, expected_message",
+        [
+            # nothing happens when the message is a plain string
+            (
+                "hello",
+                "hello",
+            ),
+            # add a context if the message is a dict
+            (
+                {"message": "hello"},
+                {"message": "hello", "ctx": 123},
+            ),
+        ],
+    )
+    def test_thread_context_logger_as_decorator(
```

Comment:
Can't you merge this with the previous test, like case A / case B and if case A you are using context, if B, the decorator?

Reply:
you could use labels to make sure the test cases can be differentiated easily